src/securityPolicy.c

```diff
@@ -768,6 +721,37 @@ static inline security_policy_t multisigWitnessPolicy(const bip44_path_t* pathSp
 	}
 }
 
+static inline security_policy_t _poolRegistrationOwnerWitnessPolicy(const bip44_path_t* path)
+{
+	switch (bip44_classifyPath(path)) {
+
+	case PATH_ORDINARY_STAKING_KEY:
+		WARN_UNLESS(bip44_isPathReasonable(path));
+		SHOW();
+		break;
+
+	default:
+		DENY();
+		break;
+	}
+}
+
+static inline security_policy_t _poolRegistrationOperatorWitnessPolicy(const bip44_path_t* path)
+{
+	switch (bip44_classifyPath(path)) {
```

Comment:
Since all the `*WitnessPolicy` functions call `bip44_classifyPath` it might make sense to call it in `policyForSignTxWitness` and pass in the path type into these functions. That way the number of calls to `bip44_classifyPath` would be reduced.

Reply:
Since the branches where bip44_classifyPath is called are disjoint, it is called only once, and we need to path the whole path as argument anyway into the policies. Whatever other minor aspects of shuffling the code around, inlining functions, grouping function calls etc. I am leaving to the compiler optimization.